### DISCUSSION TITLE
Special-case for HTTP URLs with length == 0

### DISCUSSION THREAD

**Discussion by arthuston** [pydantic/pydantic#2962] [Question]
Our app uses Optional[HttpUrl] and Optional[AnyHttpUrl] in multiple places. We use PATCH because only certain fields can be updated, but because of the limitations of PATCH, if the user wants to unset these values the best they can do is set them to the empty string, but this fails validation.

Setting the minimum length to zero fails because the protocol cannot be specified in zero characters. We're looking for alternatives that extend and/or modify the URL validators, perhaps by doing a different test if length == 0. What method(s) should we over-ride to achieve this?

**Comment by lesnik512** [comment]
If these fields are nullable in your database you can just pass `null`, then get  the dict with `.dict(exclude_unset=True)` and than save it to database
